### ISSUE TITLE
This version  model can not distinguish different text.

### ISSUE THREAD

**Issue by Wakings** [LLukas22/retsim-pytorch#3]
I use this method to extract features and calculate similarity.
But  this verision model can not distinguish different text.
```import torch
from retsim_pytorch import RETSim, RETSimConfig
from retsim_pytorch.preprocessing import binarize
config = RETSimConfig()
model = RETSim(config)

def retsim_emb(text):
    # Prepare and run inference
    binarized_inputs, chunk_ids = binarize(text)
    with torch.no_grad():
        embedded, unpooled = model(torch.tensor(binarized_inputs))
    return embedded
a = retsim_emb(["露营 小游戏 "])
b = retsim_emb(["有需要的可以问"])
ans = a@b.T
tensor([[0.9990]]) # ans

**Comment by Wakings** [comment]
The version of  unisimi  can distinguish different text.

**Comment by LLukas22** [comment]
Hm ok thats a bit weird 🤔 

Have you loaded in the safetensor weights provided [here](https://github.com/LLukas22/retsim-pytorch/blob/main/weights/model.safetensors)? 

They should be an exact copy of the [originally published weights](https://github.com/google/unisim/tree/main/unisim/embedder/models/text/retsim) from the google team.

**Comment by LLukas22** [comment]
I reran your code and with the loaded weights it produced a different output. 

```python
import torch
from retsim_pytorch import RETSim, RETSimConfig
from retsim_pytorch.preprocessing import binarize
import safetensors.torch

config = RETSimConfig()
model = RETSim(config)

# Load the .safetensors file
model_state_dict = safetensors.torch.load_file("model.safetensors")
model.load_state_dict(model_state_dict, strict=False)
model.eval()  # Set the model to evaluation mode

def retsim_emb(text):
    # Prepare and run inference
    binarized_inputs, chunk_ids = binarize(text)
    with torch.no_grad():
        embedded, unpooled = model(torch.tensor(binarized_inputs))
    return embedded
a = retsim_emb(["露营 小游戏 "])
b = retsim_emb(["有需要的可以问"])
ans = a@b.T
print(ans)
```

Results in: `tensor([[0.5207]])`

Depending on your use case you may need to reconvert your own weights, you can look at [test_converter.py](https://github.com/LLukas22/retsim-pytorch/blob/main/tests/test_converter.py) for an example of that.

**Comment by Wakings** [comment]
Thanks for your answer!
The problem is solved.
